image_data_sar/sched.py

```diff
@@ -0,0 +1,264 @@
+
+        # Special case when running a set of roll options for one obsid
+        is_roll_report = all(aca.is_roll_option for aca in acars)
+
+        label_frame = 'ACA' if aca.is_ER else 'Target'
+        context['id_label'] = f'{label_frame} roll' if is_roll_report else 'Obsid'
+
+        template_file = SPARKLES_DIR / 'index_template_preview.html'
+        template = Template(open(template_file, 'r').read())
+        out_html = template.render(context)
+
+        out_filename = report_dir / 'index.html'
+        if loud:
+            print(f'Writing output review file {out_filename}')
+        with open(out_filename, 'w') as fh:
+            fh.write(out_html)
+
+        if open_html:
+            import webbrowser
+            out_url = f'file://{out_filename.absolute()}'
+            print(f'Open URL in browser: {out_url}')
+            webbrowser.open(out_url)
+
+
+def sub_in_stars(aca):
+
+    # Sort the candidates by mag
+    aca.guides.cand_guides.sort('mag')
+    args = aca.call_args.copy()
+
+    # Sub in the faintest and stop if it still has no criticals
+    for i, n in enumerate(range(len(aca.guides.cand_guides) - 1, len(aca.guides) - 1, -1)):
+        star1 = aca.guides.cand_guides['id'][n]
+        star2 = aca.guides.cand_guides['id'][n - 1]
+        print(f"trying candidates {star1} {star2}")
+        args['include_ids_guide'] = [star1, star2]
+        naca = proseco.get_aca_catalog(**args)
+        nacar = naca.get_review_table()
+        nacar.run_aca_review()
+        if len(nacar.messages == 'critical') == 0:
+            break
+    return naca, star1, star2
+
+
+def do(load_name):
+    acas = pickle.load(gzip.open(load_name))
+    aca_arr = [acas[cat] for cat in acas]
+    aca_arr = sorted(aca_arr, key=lambda k: k.meta['date'])
+
+    # Do we want to parse the ms to check the last one?  Otherwise no duration.
+    candidates = []
+    for idx, aca in enumerate(aca_arr[0:-1]):
+        obsid = aca.meta['obsid']
+        next_aca = aca_arr[idx + 1]
+        man_dur = duration(aca.meta['att'], next_aca.meta['att'])
+        man_start = DateTime(next_aca.meta['date']).secs - man_dur
+        obs_dur = man_start - DateTime(aca.meta['date']).secs
+        if (obsid > 40000) and (obsid < 59000) and (obs_dur >= MIN_DWELL):
+            print(f"Found candidate {obsid} of dur {obs_dur} at {aca.meta['date']}")
+            candidates.append(
+                {'obsid': obsid,
+                 'dwell_end': man_start})
+
+    acars = []
+
+    for cand in candidates:
+
+        obsid = cand['obsid']
+        # Reselect stars with 16 arcsec dither
+        cargs = acas[obsid].call_args.copy()
+        cargs['dither_acq'] = (16, 16)
+        cargs['dither_guide'] = (16, 16)
+        cat = proseco.get_aca_catalog(**cargs)
+
+        ncat, star1, star2 = sub_in_stars(cat)
+
+        # Use those stars in the three observations in the splits
+        cargs['include_ids_guide'] = list(ncat.guides['id'])
```

Comment:
I think it makes the most sense to force include all 8 stars in all 3 observations (so that different temperatures and dither don't cause different star selection) but could also just check that the same 8 stars are selected.

Reply:
Agreed on forcing all stars after the initial catalog is selected.